src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/LongPollingTransport.cs

```diff
@@ -17,38 +17,66 @@ public class LongPollingTransport : IHttpTransport
         private readonly ILogger _logger;
         private readonly CancellationToken _timeoutToken;
         private readonly string _connectionId;
+        private readonly bool _isFirstRequest;
 
-        public LongPollingTransport(CancellationToken timeoutToken, PipeReader application, string connectionId, ILoggerFactory loggerFactory)
+        public LongPollingTransport(CancellationToken timeoutToken, PipeReader application, string connectionId, ILoggerFactory loggerFactory, bool isFirstRequest = false)
         {
             _timeoutToken = timeoutToken;
             _application = application;
             _connectionId = connectionId;
             _logger = loggerFactory.CreateLogger<LongPollingTransport>();
+            _isFirstRequest = isFirstRequest;
         }
 
         public async Task ProcessRequestAsync(HttpContext context, CancellationToken token)
         {
             try
             {
-                var result = await _application.ReadAsync(token);
+                if (_isFirstRequest)
+                {
+                    await context.Response.Body.FlushAsync();
+                }
+
+                ReadResult result;
+                try
+                {
+                    result = await _application.ReadAsync(token);
+                }
+                catch (Exception) when (_isFirstRequest)
```

Comment:
We can't send a 500 on the first request since the headers have already been sent so we catch here. The next poll will throw

Reply:
Put a note in the comment below about how the next poll should throw the same exception, and the client will get the error in that poll.

Reply:
What's the point in reading after flushing?

Reply:
As mentioned earlier: because we still try to write out the first batch of messages in this request, we just do it via chunked encoding.